woocommerce/woocommerce-android
# Fixes issue/3496-product-pricing-bug-fix

Closes: #3496 

## Description
This PR fixes a long standing bug in the Product detail and Variation detail screen where merchants are unable to update a product price to 0. This is possible in Core and in the iOS app. On Android, when the price is set to 0, the product shows the "Read More" button on the Shop page instead of "Add to Cart".

This PR adds logic to treat null or empty price values and 0 price values differently. 
- When the price (regular or sale price) is updated to 0, then the price in the product page should be set to 0.
- When the price (regular or sale price) is updated to an empty or null value, then the price in the product page should not display anything and the `Add to Cart` button should be hidden.
- When the regular price is set to any value and the sale price is set to 0, then the price in the product page should be displayed as 0.
- When the regular price is set to any value and the sale price is set to null, then the price in the product page should be displayed as the regular price.

## Testing instructions

1. Click on the `+` button in the Products tab and create a new product.
2. Notice that the price section displays "Add Price".
3. Click on the "Add Price" section.
4. Click on the back button.
5. Notice the the menu option is not changed in any way.
6. Click on the "Add price" section again and enter any value in the regular price field and sale price field.
7. If the sale price entered is greater than the regular price, then an error message should be displayed.
8. Enter a regular price value greater than 0 and a sale price value less than the regular price and click on the Back button.
9. Notice the new values are updated correctly. Click on the Update/Save button. View the product page and notice the values displayed correctly as entered.
10. Click on the "Add price" section again and update the regular price to 0 and leave the sale price as empty. Save the product. View the product page and notice the price is displayed as 0.
11. Click on the "Add price" section again and leave the regular price empty and leave the sale price as empty. Save the product. View the product page and notice the price is not displayed and the `Add to Cart` button is not displayed.
12. Test the above scenarios for a variation of a product.
